fastparquet/writer.py

```diff
@@ -823,10 +828,22 @@ def write(filename, data, row_group_offsets=50000000,
         is simple) or a directory containing the metadata and data-files.
     data: pandas dataframe
         The table to write
-    row_group_offsets: int or list of ints
-        If int, row-groups will be approximately this many rows, rounded down
+    row_group_offsets: int, list of ints, or str
+        - If int, row-groups will be approximately this many rows, rounded down
```

Comment:
> Is a list of absolute timestamps possible?
Yes, this is of course feasible (everything is). I directly opted for an automatic generation from `freqstr` (as per pandas wording) becuase it makes life for the user easier. This is the way I am using it.
Basically, I have several data streams with sometime redundancies.
To handle more easily the data, and to reconcile the data, splitting by time intervals is very convenient.
Even days / years after, I can update old data if I see I have a hole without rewritting full dataset.

> That might solve for those that have complex intervals. This would make it easier to append, since you can read the list of start times from the row-groups from the metadata file before append.
I don't see where it is 'easier'.
Maybe coding, and this I am not sure, because the implementation at this level is I think already 'modular: the timestamp generation is 2 functions:
- the one I implemented in `fastparquet`: `previous_date_offset`
- the other one from pandas: `date_range`

Reply:
> Question: how hard would it be to extend this to `cut` by another data type such as strings, e.g., partition 1 contains values where the index labels are A <= value <B.

Sorry, I only see your sentence here now.
IMHO, this is in such case that a concrete use case is needed, to set the limits of the feature.
My use case is a continuous stream of data, never stopping.
This is why for me, your proposal

> Is a list of absolute timestamps possible?

is not the ideal answer. Because, new timestamps are continuously added. Taking a logic for generating a `period_range` makes thing easier for using the feature. 

So to come back to your question:

> how hard would it be to extend this to cut by another data type such as strings, e.g., partition 1 contains values where the index labels are A <= value <B

Your question makes sense with your example: there are only 26 letters in the alphabet :)
Well, at some point in the code I have implemented, a list of timestamps is indeed generated, only it is localized 'around' the timestamps from the data that is about to be updated.
I guess, it would means plugging the input list from the user here, instead of generating it.

These are additional `if` to manage. For adding this complexity, it needs to serve a real use case / the demand of a user.
Does it?